task2_link_source/models/sale_order.py

```diff
@@ -0,0 +1,31 @@
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    def action_recompute_price_unit(self):
+        """
+        Action for the recompute prices button. 
+        Calls the _get_display_prices() and pass to context to differrentiate using pricelist for every sales order line
+        :param self - set of records associated to the context where this function is being called
+        """
+        for record in self:
+            # Get all the sale order lines linked to this order
+            order_lines = self.env['sale.order.line'].search([('order_id','=',record.id)])
```

Comment:
no need to search order lines this way.
sale order object has a field order_line